src/Containers/Microsoft.NET.Build.Containers/AuthHandshakeMessageHandler.cs

```diff
@@ -71,18 +71,19 @@ private static bool TryParseAuthenticationInfo(HttpResponseMessage msg, [NotNull
         if (header.Scheme is not null)
         {
             scheme = header.Scheme;
-            var keyValues = ParseBearerArgs(header.Parameter);
-            if (keyValues is null)
-            {
-                return false;
-            }
 
             if (header.Scheme.Equals(BasicAuthScheme, StringComparison.OrdinalIgnoreCase))
             {
-                return TryParseBasicAuthInfo(keyValues, msg.RequestMessage!.RequestUri!, out bearerAuthInfo);
+                bearerAuthInfo = null;
+                return true;
             }
             else if (header.Scheme.Equals(BearerAuthScheme, StringComparison.OrdinalIgnoreCase))
             {
+                var keyValues = ParseBearerArgs(header.Parameter);
+                if (keyValues is null)
+                {
+                    return false;
+                }
```

Comment:
@tmds is this the crux of the change that un-broke OpenShift? essentially not trying to force Bearer auth early?

If so, would it be possible to minimize the diff here to reduce risk for GA by removing the reordering below?

Reply:
> OpenShift

The change needed for the OpenShift issue is the removal of the `<token>` username handling.
This handling is needed only when using a credential-script handler and is already performed for us here: https://github.com/mthalman/docker-creds-provider/blob/ea7b1c9fd126e2c70c0e5bcc3e500bfe93ef8754/src/Valleysoft.DockerCredsProvider/NativeStore.cs#L59-L64.

> to reduce risk for GA by removing the reordering below?

We could only remove the `<token>` handling and preserve the ordering as it was.

We've reduced the risk for the reordering by:
- still trying the password_grant auth (unchanged) as a fallback even though the registries are expected to accept the Basic auth attempt,
- adding tests that validate the behavior by implementing the different server side auth expectations,
- and verify the implementation against different real registries.

Reply:
gotcha - so you have something like this in your docker auth file:

```json
"auths": {
  "myopenshiftregistry.mycorp.com": "base64-encoded <token>:password value"
}
```

Reply:
But the linked NativeStore code says that if the username is token that it sets password to null; so that would make the basic auth just send `PHRva2VuPg==:`.  Unless it's being loaded from some other path and doesn't go through the if you linked to?

Reply:
aha! if the `auths` are used then you use an [`EncodedStore`](https://github.com/mthalman/docker-creds-provider/blob/6e1ecd0a80755f9f0e88dc23b98b52f51a77c65e/src/Valleysoft.DockerCredsProvider/EncodedStore.cs#L6) instead of `NativeStore`. This store doesn't do the translation for `<token>` like `NativeStore` does. @bartonjs that is the missing piece for me.